src/main/java/com/aws/greengrass/lifecyclemanager/KernelLifecycle.java

```diff
@@ -462,12 +463,13 @@ public void stopAllServices(int timeoutSeconds) {
      * @param timeoutSeconds Timeout in seconds
      */
     public void softShutdown(int timeoutSeconds) {
-        logger.atDebug("system-shutdown").log("Start soft shutdown");
+        logger.atDebug(SYSTEM_SHUTDOWN_EVENT).log("Start soft shutdown");
+        stopAllServices(timeoutSeconds);
```

Comment:
Is it possible for switching the order of `stopAllServices` and `kernel.getContext().waitForPublishQueueToClear();`, `close(tlog);`, `kernel.writeEffectiveConfig();` break any existing workflow?

Reply:
We shouldn't close the tlog until services are stopped (since they might be writing to the config).

Does it matter if `waitForPublishQueueToClear()` happens before or after services are stopped?

Reply:
Going to move waitForPublishQueueToClear() before stopAllServices(), which restores the original relative order, just to be safe.